de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,343 @@
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.common.util.URI;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.APropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.model.extension.tests.model.EReferenceTest;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryAllProperty;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryComposition;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryReference;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElement;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElementOther;
+import de.dlr.sc.virsat.model.extension.tests.test.ATestConceptTestCase;
+import de.dlr.sc.virsat.model.external.tests.ExternalTestType;
+import us.hebi.matlab.mat.types.MatFile;
+
+public class MatImporterTest extends ATestConceptTestCase {
+
+	private StructuralElementInstance sei;
+	private TestStructuralElement tsei;
+	private MatExporter exporter;
+	private MatImporter importer;
+	private MatFile mat;
+	private static final int NUMBEROFELEMENTS = 6;
+
+	@Before
+	public void setUp() throws CoreException {
+		exporter = new MatExporter();
+		importer = new MatImporter();
+		super.setUp();
+		addEditingDomainAndRepository();
+		executeAsCommand(() -> loadTestConcept());
+		tsei = new TestStructuralElement(testConcept);
+		sei = tsei.getStructuralElementInstance();
+		sei.setName("testsei");
+		executeAsCommand(() -> repository.getRootEntities().add(sei));
+		mat = exporter.exportSei(sei);
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiCorrect() {
+		assertTrue("Is the same", importer.checkIfCorrectSei(sei, mat));
+		sei.setName("Testsein");
+		assertFalse("Not the same name", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongUUID() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setType(sei.getType());
+		assertFalse("Not the same uuid", importer.checkIfCorrectSei(sei2, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongType() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setUuid(sei.getUuid());
+		assertFalse("Not the same type", importer.checkIfCorrectSei(sei2, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongNumberOfChildren() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setParent(sei);
+		assertFalse("Not the same number of children", importer.checkIfCorrectSei(sei, mat));

```

Comment:
I changed it that way it is checking UUID and Type. Now children can be added and it will get the sei as a fitting one. Correct means that the sei and the one in the .mat fit to each other.